xdsl/dialects/riscv_func.py

```diff
@@ -48,9 +51,102 @@ def verify_(self):
             )
 
 
+@irdl_op_definition
+class CallOp(IRDLOperation):
+    """RISC-V function call operation"""
+
+    name = "riscv_func.call"
+    args: Annotated[VarOperand, riscv.RegisterType]
+    func_name: OpAttr[StringAttr]
+    result: Annotated[OptOpResult, riscv.RegisterType]
+
+    def __init__(
+        self,
+        func_name: StringAttr,
+        args: list[Operation | SSAValue],
+        has_result: bool = True,
+        comment: StringAttr | None = None,
+    ):
+        super().__init__(
+            operands=[args],
+            result_types=[riscv.RegisterType(riscv.Register()) if has_result else None],
+            attributes={
+                "func_name": func_name,
+                "comment": comment,
+            },
+        )
+
+
+@irdl_op_definition
+class FuncOp(IRDLOperation):
+    """RISC-V function definition operation"""
+
+    name = "riscv_func.func"
+    args: Annotated[VarOperand, riscv.RegisterType]
+    func_name: OpAttr[StringAttr]
+    func_body: SingleBlockRegion
+
```

Comment:
I feel this is missing a `function_type` style annotation? @superlopuh what's your opinion on that?